### DISCUSSION TITLE
NTC Thermistor order.

### DISCUSSION THREAD

**Discussion by zombielinux** [tagyoureit/nodejs-poolController#424] [REM] [answered]
In the wiki page on Temperature sensors for REM, there is a specified circuit. 

However, a cursory glance online shows two different potential circuits. One with the thermistor in R1 one in the R2 positions of a voltage divider. 

Is there a particular reason we can't use either position? I've got an ADC board with a built in thermistor that is in the opposite orientation of the REM documentation.

**Comment by rstrouse** [comment]
Which ADC do you have?

**Comment by rstrouse** [comment]
If you are wiring them backwards you are measuring the voltage over the resistor and not over the NTC.  Is it possible you hooked up a PTC thermistor?

**Reply by zombielinux** [reply]
Its unlikely to be a PTC. The output voltage is tracking the behavior of an inverted NTC voltage divider. 

Since a 10K NTC is paired with a 10K resistor, at the ideal crosspoint (normally 25C if memory serves), the voltage would be 1/2 VCC. Ultimately, we want to know the voltage across the thermistor, and since all other components of the voltage divider equation is known, we can easily find the answer. 

This is the math thats currently done:

Vout=Vin*R2/(R1+R2)

Vout/Vin = R2/(R1+R2)

R1Vout/Vin+R2Vout/Vin=R2

R1Vout/Vin=R2-R2Vout/Vin

R1Vout/Vin=R2(1-Vout/Vin)

R2= (R1Vout/Vin)/(1-Vout/Vin)

Some rearrangement of the second to last line gets us: 

R1=R2(1-Vout/Vin)/(Vout/Vin)=R2Vin/Vout-R2

If we deal with ADC values, its even easier, since instead of taking the raw ADC value, we just need to subtract the measured ADC value from the maximum. 

i.e. in the case of a 10 bit ADC, instead of reporting "x", we report "1024-x".

**Reply by rstrouse** [reply]
Which ADC do you have.  Is it the Alchemy one?

**Reply by zombielinux** [reply]
Its this one: https://www.amazon.com/gp/product/B07GGY7WX7

Based on the ADS1115.

**Reply by rstrouse** [reply]
Interesting.  It is hard to tell exactly how that is wired but typically the capacitor in a divider circuit like that would run to ground trough a equal resistor.  The right side of R6 is likely connected to A0.  The way to tell is to see if there is conductivity between the right side of P4 and 3v3 or the left side of R4 and 3v3.  Or test conductivity between the right side of R6 and A0.  Measuring voltage VCC decrease with increase temperature is fundamentally wrong.  The bias is not on the NTC and is on the reference circuit.  Sure there is such a thing as reverse bias voltage divider but inspecting the board it does not look that way.

**Reply by zombielinux** [reply]
![Screenshot_2022-04-22_16-44-45](https://user-images.githubusercontent.com/188401/164791172-717f3461-253c-4ae7-aa83-7bcc1504054e.png)

This is what I've been able to tone out.

**Reply by zombielinux** [reply]
And this is the full schematic. 
![Screenshot_2022-04-22_17-05-31](https://user-images.githubusercontent.com/188401/164793469-4424488f-559a-420f-98c9-3b4e9ff66161.png)

**Comment by rstrouse** [comment]
They balance the resistance across the divider.  How does that work?  It would make sense if R4 was in line with C1.  The NTC is not linear so what is the VCC.  I am not an EE, I am a software engineer so the propeller spins in the proper direction.

**Reply by zombielinux** [reply]
I'm not sure exactly what they're doing, unless the NTC they're using isn't a 10K. It would be fairly trivial to reflow a jumper to that to confirm. 

That being said, in its current configuration, its behaving appropriately, just with the inverted value. I recognize that NTC's aren't linear, but over reasonable temperature ranges, they should be close enough. The only issue I potentially see is down near 32F when you're worried about the water in the pipes freezing, but really pumps should be turned on WAY before then. 

Some simulation work shows it in the 10k range says they're biasing the resistor with an offset in order to keep the ntc in a more linear range for normal temperatures.

**Reply by rstrouse** [reply]
Alright I added a Reverse Bias checkbox to the channel.  I assume you are looking to unsolder the NTC from the board and hook up that way.  If it were me I would create a proper voltage divider and hook it up to A0.  Honestly, I am not sure they understood the concept but the reverse bias would still be accurate for linear sensors.

**Reply by Telemachus1977** [reply]
So I have the same board.  Currently not using the built in NTC, but added an external 10k one as in this diagram:

![124990826-b913db80-dff5-11eb-9ce0-80686d40bb90](https://user-images.githubusercontent.com/25725743/200331522-376641d3-8cdb-455a-bf02-51c72fb47370.png)

The values only made sense when I inverted the bias... But I have the resistor and thermistor on the opposite sides from the built in one... And even then it's about 9' C off.

**Reply by rstrouse** [reply]
Did you try a different port than A0?

**Reply by Telemachus1977** [reply]
Edit:

Now that I go back to look at it we were using port 1, in parallel with the built-in in NTC.

I have now moved the voltage divider to port 4, and the water readings are perfect without any calibration.

As far as the built-in thermistor goes, it gives pretty much the same readings whether it is biased or unbiased.

**Reply by Telemachus1977** [reply]
![Screenshot_20221107_084437_Chrome](https://user-images.githubusercontent.com/25725743/200366443-c2e5d394-f666-4fd3-8f4d-183b216f37bd.jpg)
![Screenshot_20221107_084423_Chrome](https://user-images.githubusercontent.com/25725743/200366453-abbde015-9d3a-42d6-bb0e-c9171114eeef.jpg)

Both translate to a temp value that seems too high.  Nothing in the box feels this warm.

![Screenshot_20221107_084628_Chrome](https://user-images.githubusercontent.com/25725743/200366938-f674fb26-92e4-4f7c-93a9-1becbb5f90d4.jpg)

**Reply by rstrouse** [reply]
There is a jumper on that board.  How do you have it set?

**Reply by Telemachus1977** [reply]
There is the jumper that sets the i2c address which obviously is appropriate because we are accessing it.

The other jumper essentially just turns on and off the connection to the built-in NTC.

It is in place.

Really we did not plan on using the built-in thermistor, but when tagyoureit it pointed me to this thread I realized it was there and figured we would try to get some box temperatures out of it.  I will keep it on reverse bias since that is what worked for the OP and we will try calibrating it with a different temp.  I'm not quite sure what the reference voltage should be with the resistors on both sides.

The good news is, with a regular voltage divider, Port 4 is working appropriately.

The next step is to try to use this board with a pressure transducer.

I'm still not clear if a voltage divider is needed for a pressure transducer or not.

Also we are going to try a 5 volt transducer connected to 3.3 volts. I assume the output is linear so if we can figure out the value with two different spots, we should be able to calibrate the thing.

**Comment by rstrouse** [comment, marked as answer]
If you pull REM and configure the ADC channel to have reverse bias it will calculate the range in reverse.  Still not sure what they are thinking with that though.
